### PR TITLE
Add motivation and example re. extensibility

### PR DESCRIPTION
**Pull request by rothmichaels**
I thought it might be useful to have a motivation and example about the need for library users to be able to define domain specific quantities and units with all the functionality provided by library quantitites and units.

I wrote this on the plane after CppCon when I was a bit tired, so maybe it needs an editing pass for clarity.

### CODE REVIEW

File: src/2980R0_a_motivation_scope_and_plan_for_a_physical_quantities_and_units_library.md [hunk 1]
```diff
@@ -235,6 +235,21 @@ As a result, companies either use really simple and unsafe numeric wrappers or a
 and just use `double` to express quantity values which lead to safety issues by accidentally using
 values representing the wrong quantity or having an incorrect unit.
 
+## Extensibility
+
+Many applications of a quantity and units library may need to operate on
+a combination of standard (e.g. time in seconds or distance in metres)
+and domain-specific quantities and units.  The complexity of developing
```
Comment (JohelEGP) [inline src/2980R0_a_motivation_scope_and_plan_for_a_physical_quantities_and_units_library.md:242]: For clarity, before giving examples, it should be clear that "standard" refers to "standard quantities and units".
So I suggest this rewrite:
```suggestion
a combination of quantities and units
that are standard (e.g. time in seconds or distance in metres)
and domain-specific.
The complexity of developing
```

**Review by mpusz — APPROVED** [review]
Thanks!

**Comment by mpusz** [comment]
@rothmichaels I think that I raced with @JohelEGP review while merging this PR. Could you please still consider adding the proposed changes?

**Comment by rothmichaels** [comment]
@mpusz sure thing. I'll make some updates tonight when I get back to my hotel.

> At least the storage tank example comes from https://github.com/mpusz/mp-units.
If that's the case for all others,
I guess @mpusz would ask you for a PR there
to ensure the paper's version stays up-to-date.

Do you want me to open a PR for mp-units with this new example?

**Comment by mpusz** [comment]
I think it is a great example to have in mp-units as well :-) Before submitting the PR, please check out my changes to your sample in the paper and Compiler Explorer. BTW, I am still thinking if you should use a strong dimension or a dimensionless quantities for your case. I checked that the IEC 80000 uses the latter for bits:

https://github.com/mpusz/mp-units/blob/b7e2033d33fa42843de38cfc6d0f2a4875caff4b/src/systems/iec80000/include/mp-units/systems/iec80000/quantities.h#L48

**Pull request merged**